airflow/models/dagrun.py

```diff
@@ -164,9 +164,7 @@ class DagRun(Base, LoggingMixin):
         ),
     )
 
-    task_instances = relationship(
-        TI, back_populates="dag_run", cascade="save-update, merge, delete, delete-orphan"
-    )
+    task_instances = relationship(TI, back_populates="dag_run", cascade="save-update, merge")
```

Comment:
I would prefer we keep these since we still need to delete those associated task instances when a DAG run is deleted by other means.

Reply:
Agree, I've returned cascade deletion and put the deletion of task instances before the deletion of a dag run, so cascade deletion is not affecting performance.